### DISCUSSION TITLE
How to deal with projects that do no longer publish releases

### DISCUSSION THREAD

**Discussion by jpstotz** [Homebrew/discussions#4545] [Writing Formulae/Casks]
### Output of `brew config`

```shell
-
```

### Output of `brew doctor`

```shell
-
```

### Description of issue

There are projects like [libimobiledevice](https://github.com/libimobiledevice/libimobiledevice) that have stopped releasing versions (they decided to no longer create releases, users should just use the most recent version from git). Therefore the latest version available via brew is 1.3 released in 2020 which is heavily outdated (and installing HEAD version doesn't work as the [formula is too outdated](https://github.com/libimobiledevice/libimobiledevice/issues/1217)). 

No release means no release source tar code to download and also not GIT tag you could reference. 
The only way would be to pick a git commit hash and use the GitHub archive feature to get a the repository at the time of the commit hash:

`https://github.com/libimobiledevice/libimobiledevice/archive/08aefcd597d26a5d99f64815a2b932c36f056164.tar.gz`

Is this an acceptable source for a Brew formula?

The second problem is the version. I have not tried it but I assume that brew is not able to generate a version from such an URL. So you have to specify it manually, but which version to use? As there are no releases everything is possible but what what would be an acceptable version, considering that those versions will only exists in brew... 

The last release of libimobiledevice was 1.3.0. Using git commit hashes is in my opinion a bad idea as it make sit impossible to decide which version is newer if you have two versions, so the only way would be to use  the date of the commit and pack it into a version.

Would 1.3.0-20230531 would be a suitable version for a brew formula?

**Comment by SMillerDev** [comment]
Yeah, that workflow sucks unless you're the developer of the tool I guess. What do other package managers do about this?

**Reply by jpstotz** [reply]
@SMillerDev The other repos I know like the one from Ubuntu just stays at the ancient version. Not sure if the package maintainers know at all that there will be no new releases. Not sure who people get a newer version of libimobiledevice on systems like Ubuntu. I have started to not caring about packages and directly use git clone + make install as I needed a more recent version.

**Comment by p-linnane** [comment]
Please also see this discussion: https://github.com/libimobiledevice/libimobiledevice/issues/1435

**Comment by alebcay** [comment]
> Would 1.3.0-20230531 would be a suitable version for a brew formula?

There is one notable exception to the usual tagged releases only rule, which is `luajit`. The rationale for making the exception is noted in https://github.com/Homebrew/homebrew-core/pull/99580#pullrequestreview-1022193828; I'm fairly certain the same reasons don't apply as readily here, though.
